### DISCUSSION TITLE
SwiftPM fetching in build lanes

### DISCUSSION THREAD

**Discussion by jongarate** [fastlane/fastlane#17045] [Q&A]
As of fastlane version _2.156.0_, I'm trying to separate the build process in the CI into the following lanes:
- **(1)** arrange dependencies
- **(2)** build binary
- **(3)** run tests (which builds an additional binary)

The idea being that **(1)** should take care of fetching and arranging all dependencies, currently both _SwiftPM_ and _CocoaPods_, while **(2)** and **(3)** relate to **ONLY** building and testing respectively (e.g. **NO** network activity and _ideally_ no dependency compiling).

Thing is, whenever **(2)** or **(3)** are fired _fastlane_ underlyingly seems to execute the following:
`$ xcodebuild -resolvePackageDependencies -workspace...`

I'm wondering wether it would be possible to disable the `-resolvePackageDependencies` flag and, for instance, use `-disableAutomaticPackageResolution` in it's place to avoid fetching _SwiftPM_ dependencies in every lane. I've been checking the official docs and haven't found anything related. Checking _fastlane_ source code suggests this parameter is enforced whenever we are running _XCode_ => **11**. Also tried adding an additional `xcargs` parameter to both `build_ios_app` and `scan` to no avail.

### Bonus Question
For **(1)** I'm currently firing the following command in the respective lane:

> `sh("xcodebuild", "-resolvePackageDependencies")`

Is the [spm](https://docs.fastlane.tools/actions/spm/) action a better fit for this task? I've found documentation to be scarce at best.

**Comment by jongarate** [comment]
No answer after 20d?

**Comment by bguidolim** [comment]
Hi @jongarate 

You're right, this command runs every time, because it will run anyways even without it (internally), which will cause a timeout if it takes longer than expected.

The command with `-resilvePackageDependencies` has not time out, that's why it runs before every building process, which can be via `scan`, `gym`, etc.

The good thing is, you can set the path for your dependencies (resolved on step 1), so this command will be much faster.

Does it work for you?
